configure.ac

```diff
@@ -1631,6 +1631,9 @@ AC_ARG_ENABLE(log-daemon-helpers,
 ])
 m4_include([src/log/helpers.m4])
 
+dnl Check for libraries required by ACL external helpers
+SQUID_CHECK_LIBTDB
```

Comment:
If this check is only needed for external ACL helpers, then it should be initiated inside src/acl/external/helpers.m4 (if not inside helper-specific files).

N.B. I am aware that src/acl/external/helpers.m4 is currently included unconditionally.

Reply:
I considered that, however there is no reason other helpers could not use it and no other helper-specific libraries are restricted to the `helpers.m4` file(s).

Reply:
This macro cannot be use within `required.m4` because those files are now `m4_included` conditionally based on `--enable` build settings and the `AM_CONDITIONAL` macro is not permitted to be defined conditionally.

I have moved it to `helpers.m4` and added the enforcement of once-only calls in [e7b5a3f](https://github.com/squid-cache/squid/pull/1776/commits/e7b5a3f230c9c15bf99f01afc0d2fb5f35d54cf1).

However, FYI this library and BerkleyDB are the **only** ones being checked conditional on whether the helper is included, all the others are checked first in `configure.ac` then the helper allowed/denied based on whether its `LIBFOO_LIBS` and headers are available. Removing special cases is the entire purpose of this cleanup work.

Reply:
Thank you for improving this code and addressing my primary concern.

> Removing special cases is the entire purpose of this cleanup work.

Reduction of special cases is a good thing, of course. In this context, the removal of a special case (present in two specific helpers or one helper category) means applying its correct logic to other helpers that are currently missing this correct logic. Doing so would make that (correct) logic no longer "special" but "widely used".